crates/bevy_core_pipeline/src/core_3d/main_transmissive_pass_3d_node.rs

```diff
@@ -0,0 +1,148 @@
+    type ViewQuery = (
+        &'static ExtractedCamera,
+        &'static Camera3d,
+        &'static RenderPhase<Transmissive3d>,
+        &'static ViewTarget,
+        Option<&'static ViewTransmissionTexture>,
+        &'static ViewDepthTexture,
+    );
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (camera, camera_3d, transmissive_phase, target, transmission, depth): QueryItem<
+            Self::ViewQuery,
+        >,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let view_entity = graph.view_entity();
+
+        let physical_target_size = camera.physical_target_size.unwrap();
+
+        let render_pass_descriptor = RenderPassDescriptor {
+            label: Some("main_transmissive_pass_3d"),
+            // NOTE: The transmissive pass loads the color buffer as well as overwriting it where appropriate.
+            color_attachments: &[Some(target.get_color_attachment(Operations {
+                load: LoadOp::Load,
+                store: true,
+            }))],
+            depth_stencil_attachment: Some(RenderPassDepthStencilAttachment {
+                view: &depth.view,
+                // NOTE: The transmissive main pass loads the depth buffer and possibly overwrites it
+                depth_ops: Some(Operations {
+                    load: LoadOp::Load,
+                    store: true,
+                }),
+                stencil_ops: None,
+            }),
+        };
+
+        // Run the transmissive pass, sorted back-to-front
+        // NOTE: Scoped to drop the mutable borrow of render_context
+        #[cfg(feature = "trace")]
+        let _main_transmissive_pass_3d_span = info_span!("main_transmissive_pass_3d").entered();
+
+        if !transmissive_phase.items.is_empty() {
+            let screen_space_specular_transmission_steps =
+                camera_3d.screen_space_specular_transmission_steps;
+            if screen_space_specular_transmission_steps > 0 {
+                let transmission =
+                    transmission.expect("`ViewTransmissionTexture` should exist at this point");
+
+                // `transmissive_phase.items` are depth sorted, so we split them into N = `screen_space_specular_transmission_steps`
+                // ranges, rendering them back-to-front in multiple steps, allowing multiple levels of transparency.
+                //
+                // Note: For the sake of simplicity, we currently split items evenly among steps. In the future, we
+                // might want to use a more sophisticated heuristic (e.g. based on view bounds, or with an exponential
+                // falloff so that nearby objects have more levels of transparency available to them)
+                for range in split_range(
+                    0..transmissive_phase.items.len(),
+                    screen_space_specular_transmission_steps,
+                ) {
```

Comment:
I tried to write the system to add `NoAutomaticBatching` akin to skinned and morphed meshes, however that turned out to be pretty complicated since I'd need to listen to material asset updates, and then insert/remove the component in all entities with that material handle, which seemed expensive. I also had to account for `NoAutomaticBatching` already added by the skin and morphed cases, in order to not accidentally remove them, and _when_ to run that system was also confusing to me. (Should it be `PostUpdate` like the other ones? Would the material info be already available at that stage? Should it be `Prepare`? `Render`?)

I also tried to modify the batching system to be aware of the transmission ranges, and the draw call to also have a parameter to stop batching at a specific index, however both seemed too complex/risky and I am not yet comfortable enough with the batching logic to be sure I was doing it right. After 0.12 I also want to add more control over the transmissive order, so whatever we add right now would need to be changed drastically.

Ended up going for a very simple, very low risk solution (bffc33003) where I just use an associated constant in the `PhaseItem` to disable automatic batching. If I understand it correctly, should bring zero run-time cost for phases that do support batching, since the constant will be evaluated at compile time.